src/Sarif.FunctionalTests/SarifConverterTests.cs

```diff
@@ -121,29 +167,14 @@ private void RunConverter(StringBuilder sb, string toolFormat, string inputFileN
 
             string expectedSarif = File.ReadAllText(expectedFileName);
             string actualSarif = File.ReadAllText(generatedFileName);
-
-            if (expectedSarif == actualSarif)
+            
+            if (CanGeneratedSarifBeDeserializedToExpectedSarif(actualSarif, expectedSarif))
             {
-                JsonSerializerSettings settings = new JsonSerializerSettings()
-                {
-                    ContractResolver = SarifContractResolver.Instance,
-                    Formatting = Formatting.Indented
-                };
-
-                // Make sure we can successfully deserialize what was just generated
-                SarifLog actualLog = JsonConvert.DeserializeObject<SarifLog>(actualSarif, settings);
-
-                actualSarif = JsonConvert.SerializeObject(actualLog, settings);
-
-                bool success = expectedSarif == actualSarif;
-                if (success)
-                {
-                    return;
-                }
-                else
-                {
-                    File.WriteAllText(generatedFileName, actualSarif);
-                }
+                return;
+            }
+            else
```

Comment:
I know the code was already structure this way, but while you're in here, could you please fix this:

There is no need for the `else` clause. The code can be more simply written:
```
    if (!AreEquivalentSarifLogs(actualSarif, expectedSarif)
    {
        File.WriteAllText(generatedFileName, actualSarif);

        string errorMessage = "The output of the {0} converter did not match for input {1}.";
        sb.AppendLine(string.Format(CultureInfo.CurrentCulture, errorMessage, toolFormat, inputFileName));
        sb.AppendLine("Check differences with:");
        sb.AppendLine(GenerateDiffCommand(expectedFileName, generatedFileName));
    }
}
```

Reply:
Fixed